web/src/instrumentedComponents.tsx

```diff
@@ -1,23 +1,35 @@
-import { Button, ButtonProps } from "@material-ui/core"
-import React from "react"
+import {
+  Button,
+  ButtonProps,
+  debounce,
+  TextField,
+  TextFieldProps,
+} from "@material-ui/core"
+import React, { useMemo } from "react"
 import { AnalyticsAction, incr, Tags } from "./analytics"
 
-export type InstrumentedButtonProps = ButtonProps & {
+// Shared components that implement analytics
+// 1. Saves callers from having to implement/test analytics for every interactive
+//    component.
+// 2. Allows wrappers to cheaply require uses specify analytics params.
```

Comment:
i'm not totally sure what your second note is trying to say 😅 maybe:
```suggestion
// 2. Allows wrappers to cheaply require and use specific analytics params.
```
?

Reply:
ah, was trying to use "use" as a noun, like "require uses of this to specify analytics params" or "require that things that use this specify analytics params", but yeah, that sucks